tests/RobotFramework/tests/tedge_connect/tedge_connect_test.robot

```diff
@@ -24,7 +24,8 @@ Non-root users should be able to read the mosquitto configuration files #2154
     Execute Command    sudo tedge reconnect c8y
     Should Have File Permissions    /etc/tedge/mosquitto-conf/tedge-mosquitto.conf    644 root:root
     Should Have File Permissions    /etc/tedge/mosquitto-conf/c8y-bridge.conf    644 root:root
-    Execute Command    sudo tedge config set mqtt.bridge.built_in true # Reset things after running the test
+    # Reset things after running the test
+    [Teardown]    Execute Command    sudo tedge config unset mqtt.bridge.built_in
```

Comment:
Shouldn't the teardown be registered earlier in the test (at the beginning itself or as soon as the `built_in` setting is updated), so that it is triggered even if the steps like `tedge reconnect` fails in line no. 24?

Reply:
Good catch, I missed that. I just pushed the corrected commit (I squashed the fixup immediately given the small size of the PR).

Reply:
Actually, I just double checked the RobotFramework docs, and ran a test myself, and the `[Teardown]` block is respected no matter where it is defined in the test (though this was initially an accidental placement), however I will put the Teardown step at the end again as it reads better, e.g. [Setup] -> Test steps -> [Teardown].

Here is an example where I forced an error and the `[Teardown]` is defined after the error.

```
Non-root users should be able to read the mosquitto configuration files #2154
    [Tags]    \#2154
    # Reset things after running the test
    Execute Command    sudo tedge connect c8y || true
    Should Have File Permissions    /etc/tedge/mosquitto-conf/tedge-mosquitto.conf    644 root:root
    Execute Command    sudo tedge config set mqtt.bridge.built_in false
    Execute Command    sudo tedge reconnect c8y
    Should Have File Permissions    /etc/tedge/mosquitto-conf/tedge-mosquitto.conf    666 root:root
    Should Have File Permissions    /etc/tedge/mosquitto-conf/c8y-bridge.conf    644 root:root
    [Teardown]    Execute Command    sudo tedge config unset mqtt.bridge.built_in
```

<img width="1192" alt="image" src="https://github.com/user-attachments/assets/23c53c53-d898-4041-b3d6-b485ecd9b3ff">

Reply:
So I've reverted back again (sorry for the back and forth)

Reply:
That's good to know. I just wasn't familiar with that capability of Robot framework. And yeah, that definitely reads better.